### PR TITLE
squid API v0.2: first draft

### PR DESCRIPTION
**Pull request by ssallam**
Update squid APID to v0.2
* Update agreement conditions to line up with keeper v0.8

### CODE REVIEW

**Review by aaitor — APPROVED** [review]
lgtm

**Review by eruizgar91 — COMMENTED** [review]
LGTM!

File: doc/architecture/squid.md [hunk 1]
```diff
@@ -723,27 +719,20 @@ ocean.agreements
 
 ---
 
-ocean.agreements.conditions.payment
+ocean.agreements.conditions
```
Comment (Pedro-vk) [inline doc/architecture/squid.md]: I don't totally understand that. Why do we have a module that has `lockReward`, `grantAccess`...?
This behaviour is part of the different conditions/templates, right? They are like shortcuts to work with `EscrowAccessSecretStoreTemplate` template?

Comment (ssallam) [reply doc/architecture/squid.md]: Yes, these form part of the `Access` agreement flow.

**Pull request merged**